### PR TITLE
rfr-17 Add page for loading / displaying channel item link

### PR DESCRIPTION
**Pull request by Andruhin**
Added navigation to web activity. Added reload, share and open in external browser functionality

### CODE REVIEW

File: app/src/main/java/rss/feed/reader/ui/activities/WebActivity.java [hunk 1]
```diff
@@ -1,30 +1,185 @@
...
+     */
+    private void onRefreshPage() {
+        mWebView.reload();
+    }
+
+    /**
+     * share loaded news
+     */
+    private void onShareNews() {
+        String newsUrl = mWebView.getUrl();
```
Comment (Vovchyk) [inline app/src/main/java/rss/feed/reader/ui/activities/WebActivity.java]: Consider creating NewsActivity (only for displaying news) which extends WebActivity. We may need WebActivity for displaying another web content that user should not be able to share (e.g T&C, Help or Software Licences information).

File: app/src/main/java/rss/feed/reader/ui/base/BaseActivity.java [hunk 2]
```diff
@@ -95,4 +95,8 @@ public ActivityComponent getActivityComponent() {
     }
 
     protected abstract void injectActivity(ActivityComponent activityComponent);
+
+    public void showToast(int msgId){
```
Comment (Vovchyk) [inline app/src/main/java/rss/feed/reader/ui/base/BaseActivity.java]: I'd move it into some utils class with the following signature:
public static void showToast(@NonNull Content content) {...

File: app/src/main/java/rss/feed/reader/ui/activities/WebActivity.java [hunk 3]
```diff
@@ -1,30 +1,257 @@
...
 
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
         String newsUrl = getIntent().getStringExtra(EXTRAS_NEWS_URL);
-        WebView webView = (WebView) findViewById(R.id.webView);
-        webView.setWebViewClient(new WebViewClient());
-        WebSettings webSettings = webView.getSettings();
+        isExternalBrowserEnabled = getIntent().getBooleanExtra(EXTRAS_IS_EXTERNAL_BROWSER_ENABLED, false);
+        isNavigationEnabled = getIntent().getBooleanExtra(EXTRAS_IS_NAVIGATION_ENABLED, false);
+        isSharingEnabled = getIntent().getBooleanExtra(EXTRAS_IS_SHARING_ENABLED, false);
```
Comment (Vovchyk) [inline app/src/main/java/rss/feed/reader/ui/activities/WebActivity.java]: I remember you've promised me that we would also have *title* intent parameter ;)

File: app/src/main/java/rss/feed/reader/ui/activities/WebActivity.java [hunk 4]
```diff
@@ -1,30 +1,257 @@
...
+            showForwardItem(view.canGoForward());
+        }
+    }
+
+    /**
+     * load news url to webView
+     *
+     * @param newsUrl
+     */
+    private void loadNewsUrl(String newsUrl) {
```
Comment (Vovchyk) [inline app/src/main/java/rss/feed/reader/ui/activities/WebActivity.java]: I guess *WebActivity* shouldn't know anything about news, should it? Please created a new NewsActivity for that

File: app/src/main/java/rss/feed/reader/ui/activities/WebActivity.java (hunk 3 above)
Comment (Andruhin) [inline app/src/main/java/rss/feed/reader/ui/activities/WebActivity.java]: ok, I'll add

**Pull request merged**